### PR TITLE
Supporting passing context to static rendering

### PR DESCRIPTION
**Pull request by hunterbmt**
Supporting passing context to static rendering. Not yet support .context() API

### CODE REVIEW

File: test/staticRender-spec.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import { describeWithDOM } from './_helpers';
+import { describeWithDOM, describeIf } from './_helpers';
 import React from 'react';
 import { expect } from 'chai';
 import { render } from '../src/';
+import { REACT013 } from '../src/version';
 
 describeWithDOM('render', () => {
-  describe('context', () => {
+  describeIf(!REACT013, 'context', () => {
```
Comment (Aweary) [inline test/staticRender-spec.js]: @hunterbmt it doesn't look like this `describeIf` block is wrapping all the new context tests

Comment (hunterbmt) [inline test/staticRender-spec.js]: That's correct @Aweary. I fixed it. Sorry for this basic mistake

File: src/render.js [hunk 2]
```diff
@@ -11,9 +13,27 @@ import cheerio from 'cheerio';
...
+      objectAssign(childContextTypes, options.childContextTypes);
+    }
+    class ContextWrapper extends React.Component {
+      getChildContext() {
+        return options.context;
+      }
+      render() {
+        return node;
+      }
+    }
```
Comment (Aweary) [inline src/render.js]: Any way we can avoid defining the class every time `render` is invoked? I guess its not a huge deal if it's required. Maybe we can move this logic out of the `render` method into something like `renderWithContext` to make the actual `render` call a little cleaner

```js
if (options.childContextTypes) {
  return renderWithContext(node, options)
}
```

What do you think?

Comment (lencioni) [inline src/render.js]: Yeah that might be a little nicer, but I also think that it is okay the way it currently is written.

**Comment by gerardmrk** [comment]
Hello there, is this still happening? I would really like this feature, thanks!

**Comment by ljharb** [comment]
@hunterbmt are you still interested in completing this PR?

If not, @gerardmrk, if you can point me to a commit that adds the documentation, I can cherry-pick it onto this PR, and do the rebase.

**Comment by hunterbmt** [comment]
@ljharb I will finish it today (GMT +7). Sorry for this delay, I thought I finished it already but obviously, I'm not 😿

**Comment by hunterbmt** [comment]
@ljharb could you help me to double check and merge this PR ?
Please let me know if I need to update anything.

Regards,

File: src/render.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  ContextWrapper.childContextTypes = childContextTypes;
+  return ContextWrapper;
+}
+
+export default function render(node, options = {}) {
+  if (options.context && (node.type.contextTypes || options.childContextTypes)) {
+    const childContextTypes = node.type.contextTypes || {};
+    if (options.childContextTypes) {
+      objectAssign(childContextTypes, options.childContextTypes);
```
Comment (aweary) [inline src/render.jsx]: Won't this mutate `node.type.contextTypes` if it exists?

Comment (ljharb) [inline src/render.jsx]: good catch, let's not mutate

**Comment by ljharb** [comment]
or better:
```js
const childContextTypes = objectAssign({}, node.type.contextTypes || {}, options.childContextTypes);
```

**Comment by hunterbmt** [comment]
@ljharb nice catch. I updated the PR.

**Comment by hunterbmt** [comment]
@ljharb could we merge this PR ?

**Comment by ljharb** [comment]
@hunterbmt sure, could you rebase onto latest master?

**Comment by hunterbmt** [comment]
@ljharb I already do that via github UI (you could check the last "merge" commit)

**Comment by hunterbmt** [comment]
@ljharb I rebased master into my branch. Could you help me to merged this one ?

**Comment by ljharb** [comment]
@hunterbmt would you also mind checking the "allow edits" checkbox in the right hand column?

**Comment by hunterbmt** [comment]
@ljharb Already done.

**Pull request merged**